.github/workflows/publish_charm.yaml

```diff
@@ -119,33 +108,33 @@ jobs:
         run: |
           sudo snap install charmcraft --classic
       - name: Download image artifacts
+        if: ${{ github.event_name == 'push' }}
         run: |
           RUN_IDS=$(gh api \
             --method GET \
             -H "Accept: application/vnd.github+json" \
             -H "X-GitHub-Api-Version: 2022-11-28" \
             /repos/${{ github.repository }}/actions/runs \
-            -f status=${{ needs.get-commit-sha.outputs.wf-status }} \
+            -f status=success }} \
             -f head_sha=${{ needs.get-commit-sha.outputs.sha }} \
             -f event=pull_request \
-            --jq '[.workflow_runs[] | select(.path == "${{ needs.get-commit-sha.outputs.wf-path }}") | .id]')
-          [[ $(echo $RUN_IDS | jq length) -eq 0 ]] && echo no workflow found && exit 1
+            --jq '[.workflow_runs[] | select(.path == ".github/workflows/integration_test.yaml") | .id]')
           [[ $(echo $RUN_IDS | jq length) -gt 1 ]] && echo multiple workflows found && exit 1
           RUN_ID=$(echo $RUN_IDS | jq ".[0]")
-          echo $RUN_ID
-          echo ${{ github.repository }}
-          echo ${{ matrix.image }}
           gh run download $RUN_ID -R ${{ github.repository }} -n ${{ matrix.image }}
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Download image artifacts
+        uses: actions/download-artifact@v3
+        if: ${{ github.event_name == 'pull_request' }}
+        with:
+          name: ${{ matrix.image }}
       - name: Publish image
         env:
           CHARMCRAFT_AUTH:  ${{ secrets.CHARMHUB_TOKEN }}
         run: |
-          tree
-          imageFile=${matrix.image}/${matrix.image}.tar
           charmName=$(yq '.name' metadata.yaml)
-          charmcraft upload-resource $charmName ${{ matrix.image }}-image --filepath=$imageFile --verbosity=trace
+          charmcraft upload-resource $charmName ${{ matrix.image }}-image --filepath=${{ matrix.image }}.tar --verbosity=trace
```

Comment:
I think we could reduce the verbosity before merging

Reply:
Done. Thx